lib/rules/no-side-effects-in-computed-properties.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @fileoverview Don't introduce side effects in computed properties
+ * @author Michał Sajnóg
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function parseMemberExpression (node) {
+  const members = []
+  let memberExpression
+
+  if (node.type === 'MemberExpression') {
+    memberExpression = node
+
+    while (memberExpression.type === 'MemberExpression') {
+      if (memberExpression.property.type === 'Identifier') {
+        members.push(memberExpression.property.name)
+      }
+      memberExpression = memberExpression.object
+    }
+
+    if (memberExpression.type === 'ThisExpression') {
+      members.push('this')
+    } else if (memberExpression.type === 'Identifier') {
+      members.push(memberExpression.name)
+    }
+  }
+
+  return members.reverse()
+}
+
+function getComputedProperties (componentProperties) {
+  const computedPropertiesNode = componentProperties
+    .filter(p =>
+      p.key.type === 'Identifier' &&
+      p.key.name === 'computed' &&
+      p.value.type === 'ObjectExpression'
+    )[0]
+
+  if (!computedPropertiesNode) { return [] }
+
+  const computedProperties = computedPropertiesNode.value.properties
+
+  return computedProperties.map(cp => {
```

Comment:
in case when there is only
```js
computed: {
  foo: {
    set () {}
  }
}
```
returned value is going to be 
```js
return [
  {
    key: '....',
    value: undefined
  }
]
```

And code when its used has to check for that case or we should filter it here